pepper-apis/src/main/java/org/broadinstitute/ddp/service/UserService.java

```diff
@@ -0,0 +1,76 @@
+import org.broadinstitute.ddp.db.dao.StudyGovernanceSql;
+import org.broadinstitute.ddp.db.dao.UserGovernanceSql;
+import org.broadinstitute.ddp.db.dao.UserProfileDao;
+import org.jdbi.v3.core.Handle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.Set;
+
+/**
+ * Entry-point for back-end operations related to users.
+ */
+public class UserService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(UserService.class);
+
+    public UserService() {
+    }
+
+    /**
+     * Deletes the user with specified id
+     *
+     * @param userId id of the user to be deleted
+     */
+    public void deleteUser(Handle handle, Long userId) {
+        LOG.info("Deleting user with id={}", userId);
+        // Removing user profile
+        handle.attach(UserProfileDao.class).getUserProfileSql().deleteByUserId(userId);
+        // Removing medical provider
+        handle.attach(JdbiMedicalProvider.class).deleteByUserId(userId);
+        // Removing user enrollment statuses
+        handle.attach(JdbiUserStudyEnrollment.class).deleteByUserId(userId);
+        // Removing user addresses
+        handle.attach(JdbiTempMailAddress.class).deleteTempAddressByParticipantId(userId);
+        JdbiMailAddress mailAddressDao = handle.attach(JdbiMailAddress.class);
+        mailAddressDao.deleteDefaultAddressByParticipantId(userId);
+        mailAddressDao.deleteAddressByParticipantId(userId);
+        // Removing kit requests
+        handle.attach(DsmKitRequestDao.class).deleteKitRequestsByParticipantId(userId);
+        // Removing all the answers and activity instances
+        ActivityInstanceDao instanceDao = handle.attach(ActivityInstanceDao.class);
+        Set<Long> instanceIds = instanceDao.findAllInstanceIdsByUserIds(Collections.singleton(userId));
+        LOG.info("Found {} activity instances to delete", instanceIds.size());
+        DBUtils.checkDelete(instanceIds.size(), instanceDao.deleteAllByIds(instanceIds));
+        // Removing counters, events
+        handle.attach(JdbiEventConfigurationOccurrenceCounter.class).deleteAllByParticipantId(userId);
+        JdbiQueuedEvent queueEvent = handle.attach(JdbiQueuedEvent.class);
+        queueEvent.deleteQueuedNotificationSubstitutionsByUserIds(Collections.singleton(userId));
+        queueEvent.deleteQueuedNotificationsByUserIds(Collections.singleton(userId));
+        queueEvent.deleteQueuedEventsByUserIds(Collections.singleton(userId));
+        // Removing governances
+        UserGovernanceSql userGovernanceSql = handle.attach(UserGovernanceSql.class);
+        userGovernanceSql.deleteAllGovernancesByOperatorUserId(userId);
+        userGovernanceSql.deleteAllGovernancesByParticipantUserId(userId);
+        handle.attach(StudyGovernanceSql.class).deleteAgeUpCandidateByParticipantId(userId);
+        // Removing user itself
+        handle.attach(DataExportDao.class).deleteDataSyncRequestsForUser(userId);
```

Comment:
Can you investigate deleting the user's documents in Elasticsearch as well? DSM depends on ES for user data so if we don't delete it from ES, it will not be accurate. Some thoughts:

* Should delete from the `participants` and `participants_structured` indices. Documents are identified by user guid. We should call ES's [Delete API](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/docs-delete.html) for this.
* Should also delete from `users` index. Document is also identified by user guid. HOWEVER, document has relationship between participant and proxy, and each proxy also have their own documents in here. What this means is that once the governed participant is deleted, we should update the proxy's document so the relationship is updated. We might be able to do it by queuing up `queueDataSync(proxy, study)`.
* Should lookup which study this is and use the correct indices for that study.
* You can see the structure of the documents in directory `ddp-study-server/elasticsearch/templates`.
* You can get an ES client using `ElasticsearchServiceUtil`.

Looks like it might be a decent amount of effort. If so, might be good to split into a separate ticket/PR.

Reply:
It turned out to be not a big piece of code so I kept the Elasticsearch changes in this PR.

Reply:
I'm okay with having the ES changes in this file. Now that we're using ES in other places, we should consolidate those operations into one place, but that's a separate topic for another day.